pkg/utils/overriding/merging.go

```diff
@@ -61,20 +65,23 @@ func ensureNoConflictsWithPlugins(mainContent *keyedDevWorkspaceTemplateSpecCont
 		allSpecs...)
 }
 
-func mergeKeyedDevWorkspaceTemplateSpec(
-	mainContent *keyedDevWorkspaceTemplateSpecContent,
-	parentFlattenedContent *keyedDevWorkspaceTemplateSpecContent,
-	pluginFlattenedContents ...*keyedDevWorkspaceTemplateSpecContent) (*keyedDevWorkspaceTemplateSpecContent, error) {
+// MergeDevWorkspaceTemplateSpec implements the merging logic of a main devfile content with flattened, already-overriden parent devfiles or plugins.
+// On an `main` `DevfileWorkspaceTemplateSpec` (which is the core part of a devfile, without the `apiVersion` and `metadata`),
+// it allows adding all the new overriden elements provided by flattened parent and plugins
+//
+// It is not allowed for to have duplicate (== with same key) commands, components or projects between the main content and the parent or plugins.
+// An error would be thrown
+//
+// The result is a transformed `DevfileWorkspaceTemplateSpec` object, that does not contain any `plugin` component
+// (since they are expected to be provided as flattened overriden devfiles in teh arguments)
+func MergeDevWorkspaceTemplateSpec(
```

Comment:
Since it's exported, this function should go at the top of the file.

Reply:
Fixed